radix-engine-queries/src/query/traverse.rs

```diff
@@ -175,24 +174,58 @@ impl<'s, 'v, S: SubstateDatabase, V: StateTreeVisitor> StateTreeTraverser<'s, 'v
-                        for (substate_key, substate_value) in x {
-                            let (_, owned_nodes, _) = IndexedScryptoValue::from_vec(substate_value)
-                                .expect("Substate is not a scrypto value")
-                                .unpack();
-                            for child_node_id in owned_nodes {
-                                self.traverse_recursive(
-                                    Some(&(
-                                        node_id,
-                                        SysModuleId::Object.into(),
-                                        substate_key.clone(),
-                                    )),
-                                    child_node_id,
-                                    depth + 1,
-                                );
+                        match t {
+                            SysModuleId::Object
+                            | SysModuleId::TypeInfo
+                            | SysModuleId::Royalty
+                            | SysModuleId::AccessRules => {
+                                // List all iterable modules (currently `ObjectState` & `Metadata`)
+                                let x = self
+                                    .substate_db
+                                    .list_mapped::<SpreadPrefixKeyMapper, ScryptoValue, TupleKey>(
+                                        &node_id,
+                                        t.into(),
+                                    );
+                                for (tuple_key, substate_value) in x {
+                                    let (_, owned_nodes, _) =
+                                        IndexedScryptoValue::from_scrypto_value(substate_value)
+                                            .unpack();
+                                    for child_node_id in owned_nodes {
+                                        self.traverse_recursive(
+                                            Some(&(
+                                                node_id,
+                                                t.into(),
+                                                SubstateKey::Tuple(tuple_key),
+                                            )),
+                                            child_node_id,
+                                            depth + 1,
+                                        );
+                                    }
+                                }
+                            }
+
+                            SysModuleId::Metadata | SysModuleId::Virtualized => {
+                                for (map_key, value) in self
+                                    .substate_db
+                                    .list_mapped::<SpreadPrefixKeyMapper, ScryptoValue, MapKey>(
+                                        &node_id,
+                                        SysModuleId::Virtualized.into(),
```

Comment:
previously, this line (effectively) said `t.into(),`
and right now this is the only difference between the 2 introduced "branches" (well, apart from the `TupleKey` vs `MapKey`).

Just checking - is this intentional?

Reply:
Not intentional this should be `t.into()`

Reply:
alright, I turned this duplicated code into a shared method (parameterized with `<K>` for tuple vs map aspect).